src/ray/object_manager/plasma/object_lifecycle_manager.cc

```diff
@@ -0,0 +1,281 @@
+  return object_store_->GetObject(object_id);
+}
+
+const LocalObject *ObjectLifecycleManager::SealObject(const ObjectID &object_id) {
+  // TODO(scv119): should we check delete object from eargerly_deletion_objects_?
+  return object_store_->SealObject(object_id);
+}
+
+bool ObjectLifecycleManager::AbortObject(const ObjectID &object_id) {
+  auto entry = object_store_->GetObject(object_id);
+  if (entry == nullptr) {
+    RAY_LOG(ERROR) << "To abort an object it must be in the object table.";
+    return false;
+  }
+  if (entry->state == ObjectState::PLASMA_SEALED) {
+    RAY_LOG(ERROR) << "To abort an object it must not have been sealed.";
+    return false;
+  }
+  if (entry->ref_count > 0) {
+    // A client was using this object.
+    num_bytes_in_use_ -= entry->GetObjectSize();
+    RAY_LOG(DEBUG) << "Erasing object " << object_id << " with nonzero ref count"
+                   << object_id << ", num bytes in use is now " << num_bytes_in_use_;
+  }
+
+  DeleteObjectInternal(object_id);
+  return true;
+}
+
+PlasmaError ObjectLifecycleManager::DeleteObject(const ObjectID &object_id) {
+  auto entry = object_store_->GetObject(object_id);
+  if (entry == nullptr) {
+    return PlasmaError::ObjectNonexistent;
+  }
+
+  // TODO(scv119): should we delete unsealed with ref_count 0?
+  if (entry->state != ObjectState::PLASMA_SEALED) {
+    // To delete an object it must have been sealed,
+    // otherwise there might be memeory corruption.
+    // Put it into deletion cache, it will be deleted later.
+    eargerly_deletion_objects_.emplace(object_id);
+    return PlasmaError::ObjectNotSealed;
+  }
+
+  if (entry->ref_count != 0) {
+    // To delete an object, there must be no clients currently using it.
+    // Put it into deletion cache, it will be deleted later.
+    eargerly_deletion_objects_.emplace(object_id);
+    return PlasmaError::ObjectInUse;
+  }
+
+  DeleteObjectInternal(object_id);
+  return PlasmaError::OK;
+}
+
+int64_t ObjectLifecycleManager::RequireSpace(int64_t size) {
+  std::vector<ObjectID> objects_to_evict;
+  int64_t num_bytes_evicted =
+      eviction_policy_->ChooseObjectsToEvict(size, &objects_to_evict);
+  EvictObjects(objects_to_evict);
+  return num_bytes_evicted;
+}
+
+bool ObjectLifecycleManager::AddReference(const ObjectID &object_id) {
+  auto entry = object_store_->GetObject(object_id);
+  if (!entry) {
+    RAY_LOG(ERROR) << object_id << " doesn't exist, add reference failed.";
+    return false;
+  }
+  // If there are no other clients using this object, notify the eviction policy
+  // that the object is being used.
+  if (entry->ref_count == 0) {
+    // Tell the eviction policy that this object is being used.
+    eviction_policy_->BeginObjectAccess(object_id);
+    num_bytes_in_use_ += entry->GetObjectSize();
+  }
+  // Increase reference count.
+  entry->ref_count++;
+  RAY_LOG(DEBUG) << "Object " << object_id << " in use by client"
```

Comment:
"in use by client" seems to be wrong comment here. Why don't we change it to `reference has incremented`?